From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 8 Aug 2025 19:32:47 +0200
Subject: xdp-app-info: Simplify GAppInfo creation

Since we're all in the same page now, we can just create the GAppInfo
using the app ID as base

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 src/xdp-app-info-private.h | 2 --
 src/xdp-app-info.c         | 8 +++-----
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/xdp-app-info-private.h b/src/xdp-app-info-private.h
index 55bbc12..b9f913e 100644
--- a/src/xdp-app-info-private.h
+++ b/src/xdp-app-info-private.h
@@ -54,7 +54,5 @@ struct _XdpAppInfoClass
   gboolean (*validate_dynamic_launcher) (XdpAppInfo  *app_info,
                                          GKeyFile    *key_file,
                                          GError     **error);
-
-  GAppInfo * (*create_gappinfo) (XdpAppInfo *app_info);
 };
 
diff --git a/src/xdp-app-info.c b/src/xdp-app-info.c
index 29a59f2..69914d6 100644
--- a/src/xdp-app-info.c
+++ b/src/xdp-app-info.c
@@ -77,6 +77,7 @@ typedef struct _XdpAppInfoPrivate
 } XdpAppInfoPrivate;
 
 static void g_initable_init_iface (GInitableIface *iface);
+static GAppInfo * xdp_app_info_create_gappinfo (XdpAppInfo *app_info);
 
 G_DEFINE_ABSTRACT_TYPE_WITH_CODE (XdpAppInfo, xdp_app_info, G_TYPE_OBJECT,
                                   G_ADD_PRIVATE (XdpAppInfo)
@@ -105,8 +106,7 @@ xdp_app_info_initable_init (GInitable     *initable,
   XdpAppInfo *app_info = XDP_APP_INFO (initable);
   XdpAppInfoPrivate *priv = xdp_app_info_get_instance_private (app_info);
 
-  priv->gappinfo =
-    XDP_APP_INFO_GET_CLASS (app_info)->create_gappinfo (app_info);
+  priv->gappinfo = xdp_app_info_create_gappinfo (app_info);
 
   if ((priv->flags & XDP_APP_INFO_FLAG_REQUIRE_GAPPINFO) && !priv->gappinfo)
     {
@@ -125,7 +125,7 @@ g_initable_init_iface (GInitableIface *iface)
 }
 
 static GAppInfo *
-xdp_app_info_real_create_gappinfo (XdpAppInfo *app_info)
+xdp_app_info_create_gappinfo (XdpAppInfo *app_info)
 {
   XdpAppInfoPrivate *priv = xdp_app_info_get_instance_private (app_info);
   g_autoptr(GAppInfo) gappinfo = NULL;
@@ -256,8 +256,6 @@ xdp_app_info_class_init (XdpAppInfoClass *klass)
   object_class->get_property = xdp_app_info_get_property;
   object_class->set_property = xdp_app_info_set_property;
 
-  klass->create_gappinfo = xdp_app_info_real_create_gappinfo;
-
   properties[PROP_ENGINE] =
     g_param_spec_string ("engine", NULL, NULL,
                          NULL,
